File: lib/scripts/ruby/rubocop.rb
module Ruby
  class Rubocop
    def run
      unless installed?("rubocop")
        command("gem", "install", "rubocop")
      end
      # ugh check bundle rubocop as well
      # puts test_command?("bundle exec rubocop")

      command("rubocop")
    end
  end
end

File: internal/scripts/ruby/rubocop.go
package CIScriptsRuby

import (
	c "github.com/bcaldwell/ci-scripts/internal/CIScriptsHelpers"
)

type Rubocop struct{}

func (b *Rubocop) Run() error {
	if !c.TestCommand("bundler", "exec", "rubocop", "-V") {
		c.Command("bundler", "exec", "rubocop")
		return nil
	}

	if !c.CheckBinary("rake") {
		c.Command("gem", "install", "rubocop")
	}

	c.Command("rubocop")
	return nil
}
